include/dlaf/communication/error.h

```diff
@@ -0,0 +1,46 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <cstdio>
+#include <iostream>
+#include <memory>
+
+#include <mpi.h>
+
+namespace dlaf {
+namespace internal {
+
+/// MPI provides two predefined error handlers
+///
+/// - `MPI_ERRORS_ARE_FATAL` (the default ?): causes MPI to abort
+/// - `MPI_ERRORS_RETURN`: causes MPI to return an error values instead of aborting
+///
+/// This function is only relevant when the error handler is `MPI_ERRORS_RETURN`. To set the error
+/// handler use `MPI_Errhandler_set`.
+///
+inline void mpi_call(int err, char const* file, int line) {
+  if (err != MPI_SUCCESS) {
+    std::unique_ptr<char[]> err_buff(new char[MPI_MAX_ERROR_STRING]);
+    int err_str_len;
+    MPI_Error_string(err, err_buff.get(), &err_str_len);
+    std::printf("[MPI ERROR] %s:%d: '%s'\n", file, line, err_buff.get());
+    // MPI_Abort(MPI_COMM_WORLD, err);
+    // std::abort();
```

Comment:
Are these comments to be removed? Or is it something you're willing to leave there for future reminder?

In general, I would be against having commented code.

Reply:
I hesitated here and left this commented out for now. The issue is that if we are consistent with MPI_Abort (or set a handler using it) we have to add it to DLAF_CUDA_CALL and DLAF_ASSERT_*  . The latter will make a lot of tests require MPI initialization.

I looked at Trilinos (Tpetra and Teuchos) and Boost MPI to see whether they set a termination handler and they don't. 

Suggestions on how to proceed are welcome! :)

Reply:
For MPI errors I would go for MPI_Abort.